content/updates/terralith-monolithic-terraform-architecture.md

```diff
@@ -0,0 +1,107 @@
+
+In the context of fitting a city into a single skyscraper analogy, this is like trying to fit residential areas, industrial zones, commercial districts, and corporate headquarters all into different floors of the same building. It certainly is possible, but it becomes a nightmare to manage - think about all the noise complaints from the industrial zones! And it certainly is not easy to scale when this city’s population expands.
+
+###### Collaborating in a Terralith
+Collaboration in a Terralith setup can be challenging as well. Since all the resources are provisioned with one root module, state is stored in one file. A common best practice in TF is to use [state locking](https://developer.hashicorp.com/terraform/language/state/locking), which locks the state file so only one operation can be executed at a given time. This is intended to prevent odd drift scenarios and state file corruption.
+
+Because of this practice, two engineers working on completely different infrastructure in the Terralith at the same time can find themselves unable to perform state operations concurrently.
+
+Here is a diagram of what that might look like in practice. This highlights the hit to engineering productivity that the Terralith causes:
+
+![Collaboration in a Terralith](/img/updates/terralith/terralith-collaboration.png) <!-- Made with Excalidraw -->
+
+###### State File Bloat: Plans and Applies Slow Down
+Imagine a Terralith’s state file like a single, massive spreadsheet tracking every item in a rapidly growing warehouse. Eventually, it becomes so large that comparing the current state of the resources with the state recorded in the file or updating it takes forever.
+
+In IaC, the workflow first checks all resources against the real infrastructure, then plans the changes from your infrastructure code, and finally executes the plan by applying it. Even if we are trying to modify something as minor as renaming one resource, the system must verify against every single resource in the state.
+
+It’s a domino effect because this not only slows down the development and deployment process, but also increases the vulnerability to transient errors such as credential expirations and API rate limits. At Masterpoint, we've had clients with Terralith codebases which took over 30+ minutes for simple plans and applies. As you can imagine, they often timed out or reached the API limits.
+
+![Terralith API Limit Example](/img/updates/terralith/terralith-api-limit-example.png)
+
+
+###### Blast Radius: Walking Through a Minefield
```

Comment:
For some reason, this header doesn't have padding from the above image. Can you look into that?

Reply:
Strange thing in the markdown conversion to HTML, a standalone image like this comment is an isolated <img> tag, but when there is a comment on the same line of that markdown image, it wraps it in <p><img></p> which is what gives the other images padding (consistent with the whole page since it uses <p>'s). Added a comment for that padding.